### PR TITLE
Revise BF SDE build process

### PR DESCRIPTION
**Pull request by bocon13**
General Idea:

**Option 1: Completely auto-build Stratum Docker container (and BF SDE, if needed)**
```bash
# Build SDE, kernel modules, and Stratum
docker/build-stratum-bf-container.sh ~/bf-sde-9.2.0.tgz ~/linux-4.14.49-ONL.tar.xz
# or, use pre-build SDE and kernel modules, and just build Stratum
SDE_INSTALL_TAR=~/bf-sde-9.2.0-install.tgz docker/build-stratum-bf-container.sh
```

**Option 2: Build SDE once, then build Stratum Deb locally**
```bash
# Generates ~/bf-sde-9.2.0-install.tgz (only needs to be run once)
./build-bf-sde.sh -t ~/bf-sde-9.2.0.tgz -k ~/linux-4.14.49-ONL.tar.xz
export SDE_INSTALL_TAR=~/bf-sde-9.2.0-install.tgz
bazel build //stratum/hal/bin/barefoot:stratum_bf_deb \
      --define sde_ver=$SDE_VERSION \
      --define phal_with_onlp=$WITH_ONLP
```

**Option 3: Build Stratum from local BF SDE build**
```bash
# Build (and rebuild) SDE normally
# Make sure SDE and SDE_INSTALL are set
./patch-bf-sde-install.sh # only needed once
bazel build //stratum/hal/bin/barefoot:stratum_bf_deb \
      --define sde_ver=$SDE_VERSION \
      --define phal_with_onlp=$WITH_ONLP
```

TODO
- [x] provide instructions on how to get the kernel headers (either build tar or locally)
- [x] update READMEs with new instructions

### CODE REVIEW

File: stratum/hal/bin/barefoot/build-bf-sde.sh [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+The script builds the BF SDE for Stratum.
+
+Usage: $0 [<options>]
+
+Options:
+
+    -t, --bf-sde-tar: BF SDE tarball (Required)
+    -k, --kernel-headers-tar: Linux Kernel headers tarball
+    -l, --build-local-kernel: Build kernel module for local kernel (Default: false)
+    -j, --jobs: Number of jobs for BF SDE build (Default: 4)
```
Comment (bocon13) [inline stratum/hal/bin/barefoot/build-bf-sde.sh:21]: Add -o option for output tar

File: stratum/hal/bin/bmv2/update_config.py [hunk 2]
```diff
@@ -1,145 +0,0 @@
-#!/usr/bin/env python2
```
Comment (bocon13) [inline stratum/hal/bin/bmv2/update_config.py:1]: this doesn't need to be deleted...

File: Dockerfile.build [hunk 3]
```diff
@@ -24,13 +24,17 @@ ARG JOBS
 # + software-properties-common for add-apt-repository
 # OpenNSA dependencies
 # + libelf-dev to build the kernel modules
+# BF SDE dependecies
+# + libelf-dev to build the kernel modules
+# + libbz2-dev for tar archive p4_device_config format
+# + libssl-dev for ???  FIXME(bocon)
```
Comment (bocon13) [inline Dockerfile.build:30]: TODO

File: stratum/hal/bin/barefoot/README.md [hunk 4]
```diff
@@ -5,345 +5,36 @@ Copyright 2018-present Open Networking Foundation
...
-
-`07-25 08:30:59.834213 [x86_64_accton_wedge100bf_32x] Unable to read status from file (/sys/bus/i2c/drivers/lm75/3-0048/temp1_input)`
-
-This error occurs when ONLP can not reach the BMC CPU managing the platform.
-Either the ONL image is not correctly set up or ONLP support is simply broken on
-this particular switch model. As a workaround, [BSP mode](#Running-with-BSP-or-on-Tofino-model),
-which bypasses ONLP, is available.
+**[Testing Stratum on a Barefoot Tofino based switch](./README.test.md)**
+provides some examples of how to interact with Stratum running on a
+Barefoot Tofino based switch.
```
Comment (pudelkoM) [inline stratum/hal/bin/barefoot/README.md:40]: Should we reorder this to Building -> Running -> device config -> testing?

Comment (bocon13) [reply stratum/hal/bin/barefoot/README.md:40]: Yes, that probably makes sense logically.

**Comment by bocon13** [comment]
Add --label vcs-ref=$(git log -1 --format=%H) to tag containers with label

Do we need dirty?

File: stratum/hal/lib/barefoot/barefoot.bzl [hunk 5]
```diff
@@ -2,19 +2,28 @@
...
 
 # This Starlark rule imports the BF SDE shared libraries and headers. The
-# SDE_INSTALL environment variable needs to be set, otherwise the Stratum
-# rules for barefoot platforms cannot be built.
+# SDE_INSTALL or SDE_INSTALL_TAR environment variable needs to be set,
+# otherwise the Stratum rules for barefoot platforms cannot be built.
 
 def _impl(repository_ctx):
-    if "SDE_INSTALL" not in repository_ctx.os.environ:
+    print(repository_ctx.os.environ)
```
Comment (bocon13) [inline stratum/hal/lib/barefoot/barefoot.bzl:9]: remove

**Review by pudelkoM — APPROVED** [review]
We did a quick test with the resulting binaries and found no immediate issues.

**Pull request merged**